realm/realm-library/src/main/cpp/io_realm_internal_OsObjectStore.cpp

```diff
@@ -32,6 +32,23 @@ using namespace realm::_impl;
 static_assert(io_realm_internal_OsObjectStore_SCHEMA_NOT_VERSIONED == static_cast<jlong>(ObjectStore::NotVersioned),
               "");
 
+inline static bool is_allowed_to_primary_key(JNIEnv* env, DataType column_type)
```

Comment:
PK schema tests rely on checking if a field is indexed to know if it is a PK. Because all indexed fields were PK.

Mixed that are indexable but not PK, that caused several tests to fail. This latest commit adds a check for if a field is a PK.